Client/Frontend/Home/TopSites/DataManagement/TopSitesManager.swift

```diff
@@ -104,6 +104,7 @@ class TopSitesManager: FeatureFlaggable, HasNimbusSponsoredTiles {
         let pinnedSiteCount = countPinnedSites(sites: sites)
         let totalNumberOfShownTiles = numberOfTilesPerRow * numberOfRows
         let availableSpacesCount = totalNumberOfShownTiles - pinnedSiteCount
+        guard availableSpacesCount > 0 else { return }
```

Comment:
That code changed and isn't the same as that on main. I would also think that any missing edged case should be fixed in another PR than this huge one.

Reply:
And also, if there's 0 or less available space then Google tile or sponsored tile simply won't be added. But we should still fill in the `topSites` data since otherwise the section will be empty. Coming back to my point, if there's an issue we should open another ticket/PR.